noxfile.py

```diff
@@ -92,6 +94,35 @@ def bencher_baseline(session: nox.Session) -> None:
         silent=True,
     )
 
+@nox.session(python=["3.10", "3.11"], requires=["benchmark_model-{python}"])
+def bencher_feature_branch(session: nox.Session) -> None:
```

Comment:
One last thing about the PR is that this nox task only works properly when triggered from CI, so developers should never trigger it from their own systems, which is weird. I would mention this in the docstring and suggest to change the task name to make this fact more explicit and harder to trigger accidentally. For example: 

```suggestion
def __bencher_feature_branch_CI(session: nox.Session) -> None:
```
Additionally, I would also do the same changes to the `bencher_baseline` task which shares the same behavior.